### ISSUE TITLE
invalid server kind

### ISSUE THREAD

**Issue by acuitymd-filip** [inngest/inngest-py#166]
Hi, not sure if I'm doing something wrong, but I'm trying to follow the FastAPI example from the docs here: https://www.inngest.com/docs/getting-started/python-quick-start

I have followed the steps exactly as in the docs, my `main.py` looks like this:

```python
import logging
from fastapi import FastAPI
import inngest
import inngest.fast_api

# Create an Inngest client
inngest_client = inngest.Inngest(
    app_id="fast_api_example",
    logger=logging.getLogger("uvicorn"),
)


# Create an Inngest function
@inngest_client.create_function(
    fn_id="my_function",
    # Event that triggers this function
    trigger=inngest.TriggerEvent(event="app/my_function"),
)
async def my_function(ctx: inngest.Context, step: inngest.Step) -> str:
    ctx.logger.info(ctx.event)
    return "done"


app = FastAPI()

# Serve the Inngest endpoint
inngest.fast_api.serve(app, inngest_client, [my_function])
```

here are the installed packages:

```
❯ pip freeze  
annotated-types==0.7.0
anyio==4.6.0
certifi==2024.8.30
click==8.1.7
fastapi==0.115.0
h11==0.14.0
httpcore==1.0.5
httpx==0.27.2
idna==3.10
inngest==0.4.12
jcs==0.2.1
pydantic==2.9.2
pydantic_core==2.23.4
sniffio==1.3.1
starlette==0.38.6
typing_extensions==4.12.2
uvicorn==0.30.6
```

I'm running the dev server using this command: `npx inngest-cli@latest dev -u http://localhost:8000/api/inngest --no-discovery`

and the FastAPI server like this: `(INNGEST_DEV=1 uvicorn main:app --reload)`

I keep getting this error log message:
```
INFO:     Will watch for changes in these directories: ['/Users/filip/code/it']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [66744] using StatReload
INFO:     Started server process [66746]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
ERROR:    invalid server kind: 
INFO:     127.0.0.1:64431 - "PUT /api/inngest HTTP/1.1" 200 OK
ERROR:    invalid server kind: 
INFO:     127.0.0.1:64436 - "PUT /api/inngest HTTP/1.1" 200 OK
```

<img width="749" alt="Screenshot 2024-09-23 at 4 02 05 PM" src="https://github.com/user-attachments/assets/f74e026a-4900-4bf6-9d58-3174c22e4aa0">

Am I doing something wrong?

**Comment by acuitymd-filip** [comment]
Looks like it was introduced in the latest (1.0.2) release.
Forcing the dev server to use 1.0.1 explicitly fixes the issue.

**Comment by goodoldneon** [comment]
Looks like `1.0.2` of our Dev Server is sending an empty header. We'll release a fix -- thanks for surfacing this!

**Comment by djfarrelly** [comment]
This issue should be fixed with #1793 in the `v1.0.3` release. Please re-open the issue if the new version does not fix your issue @acuitymd-filip!
